sklearn/linear_model/meson.build

```diff
@@ -22,7 +22,7 @@ foreach name: name_list
     # TODO in principle this should go in py.exension_module below. This is
     # temporary work-around for dependency issue with .pyx.tp files. For more
     # details, see https://github.com/mesonbuild/meson/issues/13212
-    depends: [linear_model_cython_tree, utils_cython_tree],
```

Comment:
My current understanding is that you only addded `cython_shared_module` to the dependencies of the custom targets doing `pyx.tp` -> `pyx`.

Do you know why this is needed only here and not in all `py.extension_module`?

Reply:
I am not sure why exactly, but locally not adding `cython_shared_module` to dependencies seems to work fine. @thomasjpfan do you confirm?

It seems like when you do `cython --shared=sklearn._cyutility` you don't need `sklearn._cyutility` to exist at build time, but probably at import time.

Reply:
Same for me, not adding `cython_shared_module` to dependencies works on CI and locally.

@matusvalo With `cython --shared=sklearn._cyutility` is it specify that module for import time and not build time?